File: spring-cloud-commons/src/main/java/com/livk/commons/http/RestClientLoadBalanceConfiguration.java
package com.livk.commons.http;

import com.livk.auto.service.annotation.SpringAutoService;
import com.livk.commons.http.annotation.EnableHttpClient;
import org.springframework.boot.autoconfigure.AutoConfiguration;
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
import org.springframework.boot.web.client.RestClientCustomizer;
import org.springframework.cloud.client.loadbalancer.LoadBalancerInterceptor;
import org.springframework.cloud.client.loadbalancer.reactive.ReactiveLoadBalancer;
import org.springframework.context.annotation.Bean;

/**
 * @author livk
 */
@AutoConfiguration
@SpringAutoService(EnableHttpClient.class)
@ConditionalOnBean(ReactiveLoadBalancer.Factory.class)
public class RestClientLoadBalanceConfiguration {

    @Bean
    public RestClientCustomizer loadbalanceRestTemplateCustomizer(LoadBalancerInterceptor loadBalancerInterceptor) {
        return builder -> builder.requestInterceptor(loadBalancerInterceptor);
    }
}

File: spring-cloud-commons/src/main/kotlin/com/livk/commons/http/RestClientLoadBalanceConfiguration.kt
package com.livk.commons.http

import com.livk.auto.service.annotation.SpringAutoService
import com.livk.commons.http.annotation.EnableRestClient
import org.springframework.boot.autoconfigure.AutoConfiguration
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.boot.web.client.RestClientCustomizer
import org.springframework.cloud.client.loadbalancer.LoadBalancerInterceptor
import org.springframework.cloud.client.loadbalancer.reactive.ReactiveLoadBalancer
import org.springframework.context.annotation.Bean

/**
 * @author livk
 */
@AutoConfiguration
@SpringAutoService(EnableRestClient::class)
@ConditionalOnBean(ReactiveLoadBalancer.Factory::class)
open class RestClientLoadBalanceConfiguration {
    @Bean
    open fun loadbalanceRestTemplateCustomizer(loadBalancerInterceptor: LoadBalancerInterceptor): RestClientCustomizer =
        RestClientCustomizer { builder -> builder.requestInterceptor(loadBalancerInterceptor) }
}
